test/elastikube_cluster_test.go

```diff
@@ -0,0 +1,256 @@
+		kubeconfig := aws.GetS3ObjectContents(t, awsRegion, ignitionS3Bucket, "kubeconfig")
+		kubeconfigPath, _ := createKubeconfig(t, exampleFolder, kubeconfig)
+		kubeconfigContext := "default"
+		kubectlOptions := k8s.NewKubectlOptions(kubeconfigContext, kubeconfigPath)
+		test_structure.SaveKubectlOptions(t, exampleFolder, kubectlOptions)
+	})
+
+	test_structure.RunTestStage(t, "validate", func() {
+		keyPair := test_structure.LoadEc2KeyPair(t, exampleFolder)
+		terraformOptions := test_structure.LoadTerraformOptions(t, exampleFolder)
+		kubectlOptions := test_structure.LoadKubectlOptions(t, exampleFolder)
+
+		testBastionHost(t, terraformOptions, keyPair)
+		testKubernetes(t, kubectlOptions, exampleFolder)
+	})
+}
+
+func configureTerraformOptions(t *testing.T, exampleFolder string, target string, uniqueID string, awsRegion string) (ret *terraform.Options) {
+
+	phase := "test"
+	project := fmt.Sprintf("k8s-%s", uniqueID)
+	keyPairName := fmt.Sprintf("test-k8s-%s", uniqueID)
+
+	ret = &terraform.Options{
+		// The path to where our Terraform code is located
+		TerraformDir: exampleFolder,
+
+		// Variables to pass to our Terraform code using -var options
+		Vars: map[string]interface{}{
+			"aws_region":             awsRegion,
+			"phase":                  phase,
+			"project":                project,
+			"key_pair_name":          keyPairName,
+			"endpoint_public_access": "true",
+		},
+	}
+
+	if target != "" {
+		ret.Targets = []string{target}
+	}
+
+	return
+}
+
+func createKubeconfig(t *testing.T, source string, kubeconfig string) (string, error) {
+	kubeconfigPath := fmt.Sprintf("%s/.test-data/kubeconfig", source)
+
+	f, err := os.Create(kubeconfigPath)
+	if err != nil {
+		return "", err
+	}
+	l, err := f.WriteString(kubeconfig)
+	if err != nil {
+		f.Close()
+		return "", err
+	}
+	fmt.Println(l, "bytes written successfully")
+	err = f.Close()
+	if err != nil {
+		return "", err
+	}
+
+	return kubeconfigPath, nil
+}
+
+func testBastionHost(t *testing.T, terraformOptions *terraform.Options, keyPair *aws.Ec2Keypair) {
+
+	bastionPublicIP := terraform.Output(t, terraformOptions, "bastion_public_ip")
+
+	// start the ssh agent
+	sshAgent := ssh.SshAgentWithKeyPair(t, keyPair.KeyPair)
+	defer sshAgent.Stop()
+
+	// We're going to try to SSH to the instance IP, using the Key Pair we created earlier. Instead of
+	// directly using the SSH key in the SSH connection, we're going to rely on an existing SSH agent that we
+	// programatically emulate within this test. We're going to use the user "ubuntu" as we know the Instance
+	// is running an Ubuntu AMI that has such a user
+	bastionHost := &ssh.Host{
+		Hostname:         bastionPublicIP,
+		SshUserName:      "ubuntu",
+		OverrideSshAgent: sshAgent,
+	}
+
+	// It can take a minute or so for the Instance to boot up, so retry a few times
+	maxRetries := 30
+	timeBetweenRetries := 5 * time.Second
+	description := fmt.Sprintf("SSH with Agent to bastion host %s", bastionPublicIP)
```

Comment:
Could consider use 

```


const (
	MaxRetries         = 30
	TimeBetweenRetries = 5 * time.Second
)

```
at the top of this file
Ref: https://play.golang.org/p/Pd-cbrXDPkQ